tritium-processor/src/main/java/com/palantir/tritium/processor/ProxyAnnotationProcessor.java

```diff
@@ -0,0 +1,426 @@
+                                            // Let's not override clone
+                                            && !method.getModifiers().contains(Modifier.NATIVE)
+                                            // Overriding finalize impacts the way objects are garbage collected, and
+                                            // can have a dramatic impact on cost.
+                                            && !Methods.isFinalize(elements, method)) {
+                                        List<MethodElements> methods = methodsByName.computeIfAbsent(
+                                                method.getSimpleName().toString(), _key -> new ArrayList<>(1));
+                                        methods.add(
+                                                new MethodElements(asMemberOf(typeElement, mirror, method), method));
+                                    }
+                                    return null;
+                                }
+                            },
+                            null);
+                }
+            }
+        }
+        List<MethodElements> instrumentedMethods = new ArrayList<>();
+        for (List<MethodElements> methods : methodsByName.values()) {
+            for (int i = 0; i < methods.size(); i++) {
+                if (isMostSpecific(i, methods)) {
+                    instrumentedMethods.add(methods.get(i));
+                }
+            }
+        }
+        if (instrumentedMethods.isEmpty()) {
+            messager.printMessage(
+                    Kind.ERROR,
+                    "Cannot generate an instrumented implementation. The annotated interface has no methods",
+                    typeElement);
+        }
+        IdentityHashMap<MethodElements, String> methodStaticFields =
+                Methods.methodStaticFieldName(instrumentedMethods, specBuilder, annotatedType);
+        for (MethodElements method : instrumentedMethods) {
+            createMethod(method, specBuilder, methodStaticFields);
+        }
+
+        specBuilder.addMethod(MethodSpec.methodBuilder("of")
+                .addTypeVariables(typeVarNames)
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .returns(annotatedType)
+                .addParameter(ParameterSpec.builder(InvocationHandler.class, HANDLER_NAME)
+                        .build())
+                .addStatement(
+                        typeVarNames.isEmpty() ? "return new $N($N)" : "return new $N<>($N)", className, HANDLER_NAME)
+                .build());
+
+        if (specBuilder.originatingElements.size() != 1) {
+            messager.printMessage(
+                    Kind.ERROR,
+                    "The generated type must have exactly one originating element: " + specBuilder.originatingElements,
+                    typeElement);
+        }
+        return JavaFile.builder(packageName, specBuilder.build())
+                .skipJavaLangImports(true)
+                .indent("    ")
```

Comment:
what does this do?

Reply:
skipJavaLangImports avoids importing things like `java.lang.Object`. Default is to include those imports to avoid ambiguity if there's a type named `Object` defined in the current package, but we  generate code to align with the style guide for the most part.

`.indent("    ")` uses our default indent, I didn't include PJF in this processor becuase it is far more expensive than the analysis/codegen part of annotation processing.